var/spack/repos/builtin/packages/opencarp/package.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from datetime import datetime
+
+from spack import *
+
+
+class Opencarp(CMakePackage):
+    """The openCARP simulation software,
+       an open cardiac electrophysiology simulator for in-silico experiments."""
+
+    homepage = "https://www.opencarp.org"
+    git = "https://git.opencarp.org/openCARP/openCARP.git"
+
+    maintainers = ['MarieHouillon']
+
+    version('7.0', commit='78da9195', submodules=False, no_cache=True, preferred=True)
+    version('master', branch='master', submodules=False, no_cache=True)
+
+    variant('carputils', default=False, description='Installs the carputils framework')
+    variant('meshtool', default=False, description='Installs the meshtool software')
+
+    # Patch removing problematic steps in CMake process
+    patch('opencarp7.patch', when='@7.0')
+
+    depends_on('git')
+    depends_on('petsc')
+    depends_on('binutils')
+    depends_on('gengetopt')
+    depends_on('pkgconfig')
+    depends_on('python@:3.8')
+    depends_on('zlib')
+    depends_on('perl')
+
+    depends_on('py-carputils')
+    depends_on('meshtool')
+    # Use specific versions of carputils and meshtool for releases
+    for ver in ['7.0']:
+        depends_on('py-carputils@oc' + ver, when='@' + ver + ' +carputils')
+        depends_on('meshtool@oc' + ver, when='@' + ver + ' +meshtool')
+
+    def cmake_args(self):
+        args = ['-DDLOPEN:STRING=ON',
+                '-DSPACK_BUILD:STRING=ON']
+        return args
```

Comment:
Since it's a `CMakePackage`, you could use `self.define()`.  Given the values are `ON`, are the types for the two options `STRING` or `BOOL`?  If the latter, you could:

```suggestion
        return [
            self.define('DLOPEN', True),
            self.define('SPACK_BUILD', True)
        ]
```

And `self.define()` will translate them to

```
        return [
            "-DDLOPEN:BOOL=ON",
            "-DSPACK_BUILD:BOOL=ON")
        ]
```

If you need 'STRING', then you could do the following:

```
        return [
            self.define('DLOPEN', 'ON'),
            self.define('SPACK_BUILD', 'ON')
        ]
```